web/src/main/scala/quasar/api/services/package.scala

```diff
@@ -0,0 +1,72 @@
+package quasar.api
+
+import quasar.Predef._
+
+import argonaut._
+import argonaut.Argonaut._
+import org.http4s.argonaut._
+
+import org.http4s.{EntityEncoder, Response}
+import org.http4s.dsl._
+import org.http4s.headers.{`Content-Disposition`, `Content-Type`}
+
+import quasar.fs.FileSystemError._
+import quasar.fs.PathError2.{InvalidPath0, PathExists0, PathNotFound0}
+import quasar.fs._
+import quasar.fp._
+
+import scalaz._, Scalaz._
+import scalaz.concurrent.Task
+import scalaz.stream.Process
+
+import pathy.Path._
+
+package object services {
+  // TODO: Polish this up
+  def fileSystemErrorResponse(error: FileSystemError): Task[Response] =
+    error.fold(
+      pathErrorResponse,
+      unknownReadHandle => BadRequest("unknow read handle"),
+      unknownWriteHandle => BadRequest("unknow write handle"),
+      partialWrite => BadRequest("Partial write"),
+      (data, str) => BadRequest("some other thing")
+    )
+
+  def pathErrorResponse(error: PathError2): Task[Response] =
+    error match {
```

Comment:
There is a `Show` instance for `PathError2` with human readable messages that we could use here for the string bits.